packages/hashi/test/mediator.spec.js

```diff
@@ -0,0 +1,266 @@
+      expect(mediator.__messageHandlers).toEqual({});
+    });
+    it('should not add a callback to the array of callbacks if nameSpace is not defined', () => {
+      const callback = jest.fn();
+      const event = 'testEvent';
+      mediator.registerMessageHandler({ event, callback });
+      expect(mediator.__messageHandlers).toEqual({});
+    });
+  });
+  describe('removeMessageHandler method', () => {
+    it('should do nothing if there are no registered handlers', () => {
+      const nameSpace = 'testNameSpace';
+      const event = 'testEvent';
+      const callback = jest.fn();
+      mediator.removeMessageHandler({ event, nameSpace, callback });
+      expect(mediator.__messageHandlers).toEqual({});
+    });
+    it('should do nothing if there are no registered handlers for that event', () => {
+      const nameSpace = 'testNameSpace';
+      const event = 'testEvent';
+      const callback = jest.fn();
+      const testData = {
+        otherEvent: [callback],
+      };
+      mediator.__messageHandlers[nameSpace] = testData;
+      mediator.removeMessageHandler({ event, nameSpace, callback });
+      expect(mediator.__messageHandlers).toEqual({
+        [nameSpace]: testData,
+      });
+    });
+    it('should do nothing if the passed in callback does not match registered handlers for that event', () => {
+      const nameSpace = 'testNameSpace';
+      const event = 'testEvent';
+      const callback = jest.fn();
+      const otherCallback = jest.fn();
+      const testData = {
+        [event]: [otherCallback],
+      };
+      mediator.__messageHandlers[nameSpace] = testData;
+      mediator.removeMessageHandler({ event, nameSpace, callback });
+      expect(mediator.__messageHandlers).toEqual({
+        [nameSpace]: testData,
+      });
+    });
+    it('should remove the passed in callback if it matches a registered handler for that event', () => {
+      const nameSpace = 'testNameSpace';
+      const event = 'testEvent';
+      const callback = jest.fn();
+      const otherCallback = jest.fn();
+      const testData = {
+        [event]: [callback, otherCallback],
+      };
+      mediator.__messageHandlers[nameSpace] = testData;
+      mediator.removeMessageHandler({ event, nameSpace, callback });
+      expect(mediator.__messageHandlers).toEqual({
+        [nameSpace]: {
+          [event]: [otherCallback],
+        },
+      });
+    });
+    it('should remove all callbacks if no callback passed in', () => {
+      const nameSpace = 'testNameSpace';
+      const event = 'testEvent';
+      const callback = jest.fn();
+      const otherCallback = jest.fn();
+      const testData = {
+        [event]: [callback, otherCallback],
+      };
+      mediator.__messageHandlers[nameSpace] = testData;
+      mediator.removeMessageHandler({ event, nameSpace });
+      expect(mediator.__messageHandlers).toEqual({
+        [nameSpace]: {
+          [event]: [],
+        },
+      });
+    });
+  });
+});
```

Comment:
So test! Much coverage! More than reviewing your code, this was a Jest tutorial for me! :)

Reply:
There was a reason I suggested it was good reading at the content dev meeting!

One nice thing I liked was that I also used Jest to do a mixture of unit and bit more 'integration-y' testing (although, obviously not full end to end). Within the mediator, and each of the shim specs, I mostly mocked and stubbed out the other functionality, but in the sandbox client, and external client tests, I ran a bit closer to real code to test out some of the mediator messaging functionality a bit better.